org.adempiere.base/src/org/compiere/util/EMail.java

```diff
@@ -560,13 +593,13 @@ public boolean isSentOK()
 	 */
 	private void dumpMessage()
 	{
-		if (m_msg == null)
+		if (m_msg == null || !log.isLoggable(Level.FINEST))

```

Comment:
this doesn't seems right, if m_msg is null, it will only trigger NPE in the code block below.

Reply:
> this doesn't seems right, if m_msg is null, it will only trigger NPE in the code block below.

The condition is OR.
If m_msg is null will execute the return below.